### PR TITLE
Cleanup before CRAN release

### PR DESCRIPTION
**Pull request by e-kotov**
@ilyabo hi! I made some changes to the package:

- implemented and documented processing of the `time` column, so that users only need to create it in datetime or date type in R and it is automatically converted to UNIX time and used in the widget

- documented the `flowmapblue` function with examples

- created vignette based on the examples from above with gifs for website and static pngs for the CRAN vignette

- created a pkgdown website for the package that can be auto-published using GitHub action (action pushed to branch too) assuming you are willing to host the website using GitHub pages at https://flowmapblue.github.io/flowmapblue.R (needs to be turned on in https://github.com/FlowmapBlue/flowmapblue.R/settings/pages and set to deploy from branch and set to `gh-pages` branch, which I already created with the pre-rendered website).

- added opengraph card (that shows up when the link is shared on social media). It will show up for https://flowmapblue.github.io/flowmapblue.R automatically, for GitHub repo link https://github.com/FlowmapBlue/flowmapblue.R/, you need to manually upload it from here https://github.com/FlowmapBlue/flowmapblue.R/blob/prep-for-CRAN-release/man/figures/opengraph-card.png into the `Social preview` section in repo settings https://github.com/FlowmapBlue/flowmapblue.R/settings

- added GitHub action for CI/CD to autocheck any future commits/PRs with R CMD Check

- cleaned up description and added metadata, as well as CRAN comments for submission

- properly bundled the example data as `ch_locations` and `ch_flows`, they are now available as exported objects in the package. **Note: would be great to reference the source of this data for completeness of the documentation.**

### CODE REVIEW

File: CITATION.cff [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+preferred-citation:
+  type: manual
+  title: Flowmap.blue widget for R
+  authors:
+  - family-names: Boyandin
+    given-names: Ilya
+    email: ilya@boyandin.me
+    orcid: https://orcid.org/0000-0001-5585-7587
+  year: '2024'
+  url: https://github.com/e-kotov/rJavaEnv
```
Comment (ilyabo) [inline CITATION.cff]:
```suggestion
  url: https://github.com/FlowmapBlue/flowmapblue.R
```

File: codemeta.json [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+          "@type": "Person",
+          "givenName": "Ilya",
+          "familyName": "Boyandin",
+          "email": "ilya@boyandin.me",
+          "@id": "https://orcid.org/0000-0001-5585-7587"
+        }
+      ],
+      "name": "Flowmap.blue widget for R",
+      "identifier": "10.32614/CRAN.package.flowmapblue",
+      "url": "https://github.com/e-kotov/rJavaEnv",
```
Comment (ilyabo) [inline codemeta.json]:
```suggestion
      "url": "https://github.com/FlowmapBlue/flowmapblue.R",
```

File: inst/CITATION [hunk 3]
```diff
@@ -0,0 +1,12 @@
+bibentry(
+  key      = "flowmapblue_r",
+  bibtype  = "Manual",
+  title    = "Flowmap.blue widget for R",
+  author   = c(
+    person("Ilya", "Boyandin", , "ilya@boyandin.me", role = c("aut", "cph"),
+           comment = c(ORCID = "0000-0001-5585-7587"))
+  ),
+  year     = "2024",
+  url      = "https://github.com/e-kotov/rJavaEnv",
```
Comment (ilyabo) [inline inst/CITATION]:
```suggestion
  url      = "https://github.com/FlowmapBlue/flowmapblue.R",
```

**Review by ilyabo — APPROVED** [review]
Thanks @e-kotov!

lgtm, I just noticed a few URL which should be changed.
I enabled google pages and uploaded the preview image.

**Comment by e-kotov** [comment]
@ilyabo sorry about the links to my other package, I will regenerate the metadata. It go in there while i was copying over some code for reuse.

**Comment by e-kotov** [comment]
Closes #2 .

**Comment by ilyabo** [comment]
No worries. I will need to change the FlowmapBlue project config (namely convert it to a monorepo and publish a library, right now it's a nextjs app). Will try to find time for it.

**Comment by e-kotov** [comment]
@ilyabo 



> I enabled google pages and uploaded the preview image.

great! Final thing to do is to add https://flowmapblue.github.io/flowmapblue.R/ (or just enable automatic URL to gh-pages) in the front page repo settings. Like so:

Screenshot 2024-08-30 at 14 18 34


I will also add this gh-pages link to package readme and metadata and submit to CRAN.

**Pull request merged**